gcloud-java-contrib/gcloud-java-nio/src/test/java/com/google/gcloud/storage/contrib/nio/ITGcsNio.java

```diff
@@ -0,0 +1,360 @@
+      byte[] wantBytes = FILE_CONTENTS.get(0).getBytes(UTF_8);
+      for (int i = 0; i < wantBytes.length; i++) {
+        assertThat(gotBytes[i] == wantBytes[i]).isTrue();
+      }
+
+    } finally {
+      // let's not leave files around
+      Files.deleteIfExists(path);
+    }
+  }
+
+  @Test
+  public void testCreateAndWrite() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(PREFIX + randomSuffix());
+    try {
+      // file shouldn't exist initially (see above).
+      assertThat(Files.exists(path)).isFalse();
+
+      Files.createFile(path);
+      Files.write(path, FILE_CONTENTS, UTF_8);
+      // now it does.
+      assertThat(Files.exists(path)).isTrue();
+
+      // let's check that the contents look OK.
+      ByteBuffer buf = ByteBuffer.allocate(100);
+      SeekableByteChannel chan = Files.newByteChannel(path, StandardOpenOption.READ);
+      reallyRead(chan, buf);
+      byte[] gotBytes = buf.array();
+      byte[] wantBytes = FILE_CONTENTS.get(0).getBytes(UTF_8);
+      for (int i = 0; i < wantBytes.length; i++) {
+        assertThat(gotBytes[i] == wantBytes[i]).isTrue();
+      }
+
+    } finally {
+      // let's not leave files around
+      Files.deleteIfExists(path);
+    }
+  }
+
+  @Test
+  public void testWriteOnClose() throws Exception {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(PREFIX + randomSuffix());
+    // file shouldn't exist initially (see above)
+    assertThat(Files.exists(path)).isFalse();
+    try {
+      try (SeekableByteChannel chan = Files.newByteChannel(path, StandardOpenOption.WRITE)) {
+        // writing lots of contents to defeat channel-internal buffering.
+        for (int i=0; i<9999; i++) {
+          for (String s : FILE_CONTENTS) {
+            chan.write(ByteBuffer.wrap(s.getBytes(UTF_8)));
+          }
+        }
+        try {
+          Files.size(path);
+          // we shouldn't make it to this line. Not using thrown.expect because
+          // I still want to run a few lines after the exception.
+          assertThat(false).isTrue();
+        } catch (NoSuchFileException nsf) {
+          // that's what we wanted, we're good.
+        }
+      }
+      // channel now closed, the file should be there and with the new contents.
+      assertThat(Files.exists(path)).isTrue();
+      assertThat(Files.size(path)).isGreaterThan(0L);
+    } finally {
+      Files.deleteIfExists(path);
+    }
+  }
+
+  @Test
+  public void testCopy() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path src = testBucket.getPath(SML_FILE);
+    Path dst = testBucket.getPath(PREFIX + randomSuffix());
+    try {
+      // file shouldn't exist initially (see above).
+      assertThat(Files.exists(dst)).isFalse();
+
+      Files.copy(src, dst);
+
+      assertThat(Files.exists(dst)).isTrue();
+      assertThat(Files.size(dst)).isEqualTo(SML_SIZE);
```

Comment:
If you captured `SML_FILE` content as suggested above you can verify the content.